packages/core/src/lib/utils.ts

```
@@ -354,10 +354,12 @@ export const withDefaultAffiliateFees = (
 export const getAddresses = async (
   blockchainType: BlockchainType,
   htlcs: GardenHTLCModules,
+  redeemServiceEnabled: boolean,
+  addresses?: Partial<Record<BlockchainType, string>>,
 ) => {
-  // if (addresses && addresses[blockchainType]) {
-  //   return Ok(addresses[blockchainType]!);
-  // }
+  if (redeemServiceEnabled && addresses && addresses[blockchainType]) {
+    return Ok(addresses[blockchainType]!);
```

don't use ! operator, please throw an error!